test/client/subscribe.ts

```diff
@@ -0,0 +1,109 @@
+import { assert } from "chai";
+
+import rippled from "../fixtures/rippled";
+import setupClient from "../setupClient";
+
+describe("Subscription", function () {
+  beforeEach(setupClient.setup);
+  afterEach(setupClient.teardown);
+
+  it("Successfully Subscribes", async function () {
+    this.mockRippled.addResponse("subscribe", rippled.subscribe.success);
+
+    const result = await this.client.request({
+      command: "subscribe",
+    });
+
+    assert.equal(result.status, "success");
+  });
+
+  it("Successfully Unsubscribes", async function () {
+    this.mockRippled.addResponse("unsubscribe", rippled.unsubscribe);
+
+    const result = await this.client.request({
+      command: "unsubscribe",
+    });
+
+    assert.equal(result.status, "success");
+  });
+
+  it("Emits transaction", async function (done) {
+    this.client.on("transaction", (tx) => {
+      assert(tx.type === "transaction");
+      done();
+    });
+
+    this.client.connection.onMessage(
+      JSON.stringify(rippled.streams.transaction)
+    );
+  });
```

Comment:
Given that you're doing this for all the types, you could make this a helper function

Reply:
Yeah, but I'd still have to call `onMessage` w/ the appropriate `rippled.streams` fixture, which would be a lil trickier. What specifically would you be looking for here?

IMO I like having them all separate, that way you can see exactly which one broke.

Reply:
I was thinking something similar to what we're doing with addresses for the client tests - you can have an `it` in the loop, so you can still tell which one broke (because they're individual tests).